src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/DepartureReasonEntity.kt

```diff
@@ -34,9 +39,21 @@ data class DepartureReasonEntity(
   val isActive: Boolean,
   val serviceScope: String,
   val legacyDeliusReasonCode: String?,
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "parent_reason_id")
+  val parentReasonId: DepartureReasonEntity?,
 ) {
   override fun toString() = "DepartureReasonEntity:$id"
 
+  fun generateParentChildName(reason: DepartureReasonEntity?): NamedId? =
+    reason?.let { reason ->
+      reason.parentReasonId?.let {
+        parentReasonId?.let { parentReason ->
+          NamedId(id = reason.id, name = "${parentReason.name} - ${reason.name}")
+        }
+      } ?: NamedId(reason.id, reason.name)
```

Comment:
I don't think API types should be 'leaking' into the domain model. Could this return the name and the receiver wrap it up on `NamedId`?